### PR TITLE
Icon location

### PR DESCRIPTION
**Pull request by Ongy**
Adds the possibility to specify the icon location as style option.

Example screenshots:
bottom
left
right
top

### CODE REVIEW

**Comment by fourstepper** [comment]
Love it!

File: render.c [hunk 1]
```diff
@@ -228,19 +259,46 @@ static int render_notification(cairo_t *cairo, struct mako_state *state,
...
-			(text_x - icon->width) / 2;
-		double ypos = offset_y + style->border_size +
-			(notif_height - icon->height - border_size) / 2;
+		double xpos = -1;
+		double ypos = -1;
+		switch (style->icon_location) {
+		case MAKO_ICON_LOCATION_LEFT:
+			xpos = offset_x + style->border_size +
+				(text_x - icon->width) / 2;
+			ypos = offset_y + style->border_size +
```
Comment (emersion) [inline render.c:268]: Can we make this a little bit easier to read by storing this value in a `ypos_center` variable before the switch? Same for `xpos_center`.

File: render.c [hunk 2]
```diff
@@ -98,30 +98,50 @@ static int render_notification(cairo_t *cairo, struct mako_state *state,
 	int padding_height = style->padding.top + style->padding.bottom;
 	int padding_width = style->padding.left + style->padding.right;
 	int radius = style->border_radius;
+	// I like micro optimizatins
+	bool icon_vertical = style->icon_location & 0x02;
```
Comment (emersion) [inline render.c:102]: Yeah, let's not

File: include/config.h [hunk 3]
```diff
@@ -20,13 +20,21 @@ enum mako_sort_criteria {
 	MAKO_SORT_CRITERIA_URGENCY = 2,
 };
 
+enum mako_icon_location {
+	MAKO_ICON_LOCATION_LEFT = 0,
```
Comment (emersion) [inline include/config.h:24]: Nit: no need to define the values explicitly.

**Comment by Ongy** [comment]
The `ypos_center` and `xpos_center` could be assigned to the actual `xpos/ypos` as well, but I think it's slightly more semantic this way.

**Pull request merged**